### PR TITLE
Add text to test case search filter

### LINKED ISSUE #1209: Search TestCase Text - There is no option
There is no option searching TestCase Text:
image

This is basic.

### PR DESCRIPTION
**Pull request by schwarzkrieger**
Closes #1209

### CODE REVIEW

File: tcms/testcases/models.py [hunk 1]
```diff
@@ -188,6 +188,9 @@ def list(cls, query, plan=None):
         if query.get('is_automated'):
             queryset = queryset.filter(is_automated=query['is_automated'])
 
+        if query.get('text'):
+            queryset = queryset.filter(Q(text__icontains=query['text']))
+
```
Comment (atodorov) [inline tcms/testcases/models.py]: This is absolutely wrong and not used byt the search functionallity, revert it.

File: tcms/testcases/templates/testcases/search.html [hunk 2]
```diff
@@ -113,6 +113,11 @@
                 <input id="id_tag" type="text" class="form-control" value="{{ form.tag__name__in.value|default:'' }}">
                 <p class="help-block">{% trans "Separate multiple values with comma (,)" %}</p>
             </div>
+
+            <label class="col-md-1 col-lg-1" for="id_text">{% trans "Text" %}</label>
+            <div class="col-md-3 col-lg-3">
+                <input id="id_text" type="text" class="form-control" value="">
```
Comment (atodorov) [inline tcms/testcases/templates/testcases/search.html]: Either use `placeholder` attribute if you want to give a hint to the user, or remove empty value attribute.

File: tcms/testcases/forms.py [hunk 3]
```diff
@@ -118,6 +118,7 @@ class BaseCaseSearchForm(forms.Form):
     items_per_page = forms.ChoiceField(
         required=False,
         choices=ITEMS_PER_PAGE_CHOICES)
+    text = forms.CharField(required=False)
```
Comment (atodorov) [inline tcms/testcases/forms.py]: FTR not all of these fields (SearchCaseForm technically) are used to render to form on the FE. The query itself is done via JS. 

I'm not even sure we need these forms but this can be tacked later b/c there is a TODO comment for them.

File: tcms/testcases/templates/testcases/search.html (hunk 2 above)
Comment (schwarzkrieger) [reply tcms/testcases/templates/testcases/search.html]: I'll remove it to avoid problems with the translation. There is no phrase like "Enter the text" in the translation files, and "Text" is the title of the field.

File: tcms/testcases/forms.py (hunk 3 above)
Comment (schwarzkrieger) [reply tcms/testcases/forms.py]: I'll test and eventually remove this addition.

File: tcms/testcases/static/testcases/js/search.js [hunk 4]
```diff
@@ -59,6 +59,10 @@ $(document).ready(function() {
                 params['is_automated'] = false;
             };
 
+            if ($('#id_text').val()) {
+                params['text__icontains'] = $('#id_text').val();
```
Comment (asankov) [inline tcms/testcases/static/testcases/js/search.js]: I think it would be better if it is:
```javascript
const text = $('#id_text').val()
if (text) {
     params['text_icontains'] = text
}
```
this will be faster, because it calls `$` once , hence you won't go to the DOM, search for the element, etc. twice

Comment (schwarzkrieger) [reply tcms/testcases/static/testcases/js/search.js]: OK, but this might be visually inconsistent with the other part of the code. I can change it if @atodorov wouldn't mind.

Comment (atodorov) [reply tcms/testcases/static/testcases/js/search.js]: Toni is right to propose this change. It should be the canonical way since you don't want to evaluate something right after you had evaluated it.  I suspect some tools may start to complain over time about too many variable in this function but we'll deal with that when we have to.

File: tcms/testcases/static/testcases/js/search.js [hunk 5]
```diff
@@ -15,40 +15,49 @@ $(document).ready(function() {
         ajax: function(data, callback, settings) {
             var params = {};
 
-            if ($('#id_summary').val()) {
-                params['summary__icontains'] = $('#id_summary').val();
+            const summary = $('#id_summary').val();
+            if (summary) {
+                params['summary__icontains'] = summary;
```
Comment (atodorov) [inline tcms/testcases/static/testcases/js/search.js]: Do not make unrelated changes b/c it makes this PR harder to review and is out of scope. 

Revert the last commit and fix only the original issues commented in previous code review.
